pkg/pillar/cmd/zedrouter/ipam.go

```diff
@@ -49,7 +49,16 @@ func (z *zedrouter) generateAppMac(adapterNum int, appStatus *types.AppNetworkSt
 	h.Write(netInstStatus.UUIDandVersion.UUID[:])
 	nums := make([]byte, 2)
 	nums[0] = byte(adapterNum)
-	nums[1] = byte(appStatus.AppNum)
+	if appStatus.MACGenerator != types.MACGeneratorClusterDeterministic {
+		// Skipped for MACGeneratorClusterDeterministic.
+		// Appending AppNum into the hash calculation could result in cluster
+		// nodes generating different MAC address for the same app interface.
+		// This is because the order of application config processing can
+		// differ between nodes.
+		// This is undesirable. When application is rescheduled to another node,
+		// the aim is to preserve MAC addresses of its interfaces.
```

Comment:
Mmm... Am I right that when we talk about App rescheduling, we mean changing the node on which the application runs? But will it not create routing problems if we preserve the MAC address? Several entities with the same MAC sound dangerous. Shouldn't we just make an ARP announcement of the new MAC for a given IP when the application is rescheduled?